File: js/sorting/bubblesort.js
function bubbleSort(arr) {
    while(true) {
        let trocou = false
        for (let i=arr.length-1; i>0; i--) {
            if (arr[i] < arr[i -1]) {
                [arr[i], arr[i-1]] = [arr[i-1], arr[i]]
                trocou = true
            }
            console.log(i, "=>", arr)
        }
        if (!trocou) break;
    }
    return arr
}

arr = [5, 3, 8, 2, 1, 4]

console.log(bubbleSort(arr))

File: ruby/sorting/bubblesort.rb
def bubble_sort(arr)
  trocou = true
  while trocou do
    trocou = false
    i = arr.length - 1
    while i > 0 do
      if arr[i] < arr[i - 1] && i > 0
        arr[i], arr[i - 1] = arr[i - 1], arr[i]
        trocou = true
      end
      i = i - 1
      puts "#{i} => #{arr}"
    end
  end
  return arr
end

arr = [5, 3, 8, 2, 1, 4]

p bubble_sort(arr)
